external-crates/move/crates/move-analyzer/prettier-move/tests/functions/expression.move

```diff
@@ -0,0 +1,217 @@
+// options:
+// printWidth: 40
+
+module prettier::expression {
+    fun assign_expression() {
+        a = 1;
+        *a = 10;
+        *&a = 100;
+        a = (copy b);
+        b = (move c);
+        *df::borrow_mut<A, B>() = another_call();
+        (a, b) = (1, 2);
+
+        strategy.underlying_nominal_value_usdc = strategy.underlying_nominal_value_usdc - to_withdraw;
+    }
+
+    fun identified_expression() {
+        'a: { call_something() };
+    }
+
+    fun name_expression() {
+        name::expression;
+        expression<T>;
+    }
+
+    fun call_expression() {
+        call();
+        call(1, 2, 3, 4, 5);
+        call(Struct {}, Struct {});
+        call(Struct { a: 1, b: 2, c: 3 });
+        call(Struct { a: 1 }, Struct { a: 1 });
+        call(
+            nested_call(deeply_nested_call()),
+            haha(),
+        );
+        call(nested_call(LongStruct { a: 1 }));
+    }
+
+    fun call_expression_type_arguments() {
+        call<InnerTypeArguments>();
+        call<InnerTypeArguments, InnerTypeArguments>();
+
+        module::another_call<Type>(
+            module::another_call<A>(),
+            module::another_call<B>(),
+            module::another_call<C>(),
+        );
+
+        module::another_call<TypeArgument>(
+            module::another_call<A>(),
+            module::another_call<B>(),
+            module::another_call<C>(),
+        );
+    }
+
+    fun binary_expression() {
+        lhs != rhs;
+        lhs == rhs;
+        lhs || rhs;
+        lhs && rhs;
+        lhs < rhs;
+        lhs <= rhs;
+        lhs > rhs;
+        lhs >= rhs;
+        lhs << rhs;
+        lhs >> rhs;
+        lhs + rhs;
+        lhs - rhs;
+        lhs * rhs;
+        lhs / rhs;
+        lhs % rhs;
+        lhs & rhs;
+        lhs | rhs;
+        lhs ^ rhs;
+    }
+
+    fun binary_expression_folding() {
+        // binary_expression
+        say_something_really_long && say_something_really_long || say_something_really_long;
```

Comment:
Please add tests for parenthesized variants

Reply:
Acknowledged. As suggested in the other [comment](https://github.com/MystenLabs/sui/pull/18405#discussion_r1681526608), let's add some additional explanation and a test, and move on

Reply:
It would be really good to get a bit more detailed explanation on the difficulty related to binary expressions, even if they are not fully handled as part of this PR, so that we know what's missing (if anything) and what kind of effort could be involved in getting this to completion